lib/surface/compiler.ex

```diff
@@ -380,7 +435,13 @@ defmodule Surface.Compiler do
     end
   end
 
-  defp convert_node_to_ast(:component, {name, attributes, children, node_meta}, compile_meta) do
+  defp convert_node_to_ast(
+         :component,
+         {name, attributes, children, node_meta} = node,
+         compile_meta
+       ) do
+    maybe_warn_on_deprecated_if_notation(node, compile_meta)
```

Comment:
we could also put this warning in `Surface.Constructs.If.transform/1` - that will always be called when `If` is used

Reply:
It is what I would have liked to do, but we do not have access to `compile_meta` in `transform/1` so it means I would be able to use `IOHelper.warn/3`. I could create a `%CompileMeta{}` struct with the `node.meta`. What do you think is the best?

Reply:
node.meta should have all of the needed information on it

Reply:
Thanks, @lnr0626, I did see the `line` but not the `caller` :facepalm: